std/typecons.d

```diff
@@ -3875,14 +3875,118 @@ unittest
 }
 
 /**
+ * Determines if the $(D Source) type satisfies all interface requirements of
+ * $(D Targets)
+ */
+private template isConceptWith(Source, Targets...)
+if (Targets.length >= 1 && allSatisfy!(isMutable, Targets))
+{
+    import std.typetuple : staticMap;
+
+    // strict upcast
+    auto isConceptWith()() @safe pure nothrow
+    if (Targets.length == 1 && is(Source : Targets[0]))
+    {
+        return true;
+    }
+    // structural upcast
+    template isConceptWith()
+    if (!allSatisfy!(Bind!(isImplicitlyConvertible, Source), Targets))
+    {
+        auto isConceptWith()
+        {
+            return hasRequireMethods!();
+        }
+
+        // list of FuncInfo
+        alias TargetMembers = UniqMembers!(ConcatInterfaceMembers!Targets);
+        // list of function symbols
+        alias SourceMembers = GetOverloadedMethods!Source;
+
+        // Check whether all of SourceMembers satisfy covariance target in
+        // TargetMembers
+        template hasRequireMethods(size_t i = 0)
+        {
+            static if (i >= TargetMembers.length)
+                enum hasRequireMethods = true;
+            else
+            {
+                enum hasRequireMethods =
+                    findCovariantFunction!(TargetMembers[i], Source, SourceMembers) != -1 &&
+                    hasRequireMethods!(i + 1);
+            }
+        }
+    }
+}
+/// ditto
+private template isConceptWith(Source, Targets...)
+if (Targets.length >= 1 && !allSatisfy!(isMutable, Targets))
+{
+    import std.typetuple : staticMap;
+
+    alias isConceptWith = .isConceptWith!(Source, staticMap!(Unqual, Targets));
+}
+
+unittest {
+    interface Foo {
+        void foo();
+    }
+    interface Bar {
+        void bar();
+    }
+    interface FooBar : Foo, Bar {
+        void foobar();
+    }
+
+    struct A {
+        void foo() {}
+    }
+    struct B {
+        void bar() {}
+        void foobar() {}
+    }
+    class C {
+        void foo() {}
+        void bar() {}
+    }
+    struct D {
+        void foo() {}
+        void bar() {}
+        void foobar() {}
+    }
+    // Implements interface
+    static assert(isConceptWith!(A, Foo));
```

Comment:
Why is there a need to write "implements interface" to explain the opaque term "isConceptWith"? Just call it `implementsInterface` and done!